src/core/tsi/ssl_transport_security.h

```diff
@@ -165,6 +167,22 @@ tsi_result tsi_ssl_client_handshaker_factory_create_handshaker(
     tsi_ssl_client_handshaker_factory* self, const char* server_name_indication,
     tsi_handshaker** handshaker);
 
+/* Creates a TLS client TSI handshaker.
+  - server_name_indication indicates the name of the server the client is
+    trying to connect to which will be relayed to the server using the SNI
+    extension.
+  - session_cache: SSL session cache for sessions resumption.
+  - options: TLS credential options instance.
+  - factory: address of client handshaker factory to be created.
+  - handshaker: address of the handshaker pointer to be created.
+  This method returns TSI_OK on success or TSI_INVALID_PARAMETER in the case
+  where a parameter is invalid. Note that some fields of handshaker will be
+  populated once credential reload is finished. */
+tsi_result tls_tsi_client_handshaker_create(
+    const char* server_name_indication, tsi_ssl_session_cache* session_cache,
+    const grpc_tls_credentials_options* options,
```

Comment:
What about cipher suites, ALPN, SNI? It somehow needs to be plumbed. TSI does not know these values. These values are from gRPC layer.

Reply:
In the current `ssl_security_connector` implementation, the information about cipher suites and ALPN are obtained via  invoking `grpc_get_ssl_cipher_suites()` and `grpc_fill_alpn_protocol_strings`, and I don't see a problem calling those functions in the TSI layer. Also, SNI information is passed in via `server_name_indication`.  HDYT?

Reply:
Sorry about SNI. I did not look carefully.
TSI implementation supposes to be self-contained and parameter can be configurable by different caller of TSI. For example, envoy or other callers of TSI may have different set of cipher suites requirements. If TSI inside calls grpc_get_ssl_cipher_suites(), that means other users of TSI will be implicitly change cipher suites when gRPC changes cipher suites.